### PR TITLE
refactor(build): remove ssh sock from conveyor and add default agent 

### PR DESCRIPTION
**Pull request by iapershin**
This removes ssh socket from conveyor and adding default ssh agent to dockerfile builds to align behaviour between stapel and dockerfile builds

### CODE REVIEW

File: docs/pages_en/usage/build/images.md [hunk 1]
```diff
@@ -189,6 +189,22 @@ RUN --mount=type=secret,id=plainSecret \
...
+
+Example:  
+```Dockerfile
+FROM alpine
+RUN apk add --no-cache openssh-client
+RUN mkdir -p -m 0700 ~/.ssh && ssh-keyscan gitlab.com >> ~/.ssh/known_hosts
+RUN --mount=type=ssh ssh -q -T git@gitlab.com 2>&1 | tee /hello
+```
+
+You can find more details about using the SSH agent in werf [here]({{ "docs/pages_ru/usage/build/process.html#Using-the-SSH-agent-in-build-instructions" | true_relative_url }}).
```
Comment (alexey-igrychev) [inline docs/pages_en/usage/build/images.md:206]:
```suggestion
You can find more details about using the SSH agent in werf [here]({{ "usage/build/process.html#Using-the-SSH-agent-in-build-instructions" | true_relative_url }}).
```

File: docs/pages_ru/usage/build/images.md [hunk 2]
```diff
@@ -189,6 +189,23 @@ RUN --mount=type=secret,id=plainSecret \
     export WERF_BUILD_SECRET="$(cat /run/secrets/plainSecret)"
 ```
 
+
```
Comment (alexey-igrychev) [inline docs/pages_ru/usage/build/images.md:192]:
```suggestion
```

File: docs/pages_ru/usage/build/images.md [hunk 3]
```diff
@@ -189,6 +189,23 @@ RUN --mount=type=secret,id=plainSecret \
...
+Пример:
+```Dockerfile
+FROM alpine
+RUN apk add --no-cache openssh-client
+RUN mkdir -p -m 0700 ~/.ssh && ssh-keyscan gitlab.com >> ~/.ssh/known_hosts
+RUN --mount=type=ssh ssh -q -T git@gitlab.com 2>&1 | tee /hello
+```
+
+Подробнее о работе SSH-агента в werf описано [здесь]({{ "docs/pages_ru/usage/build/process.html#Использование-SSH-агента-при-сборке" | true_relative_url }}).
```
Comment (alexey-igrychev) [inline docs/pages_ru/usage/build/images.md:207]:
```suggestion
Подробнее о работе SSH-агента в werf описано [здесь]({{ "/usage/build/process.html#Использование-SSH-агента-при-сборке" | true_relative_url }}).
```

File: pkg/ssh_agent/ssh_agent.go [hunk 4]
```diff
@@ -193,7 +198,14 @@ func runSSHAgentWithKeys(ctx context.Context, keys []sshKey) (string, error) {
 }
 
 func runSSHAgent(ctx context.Context) (string, error) {
-	sockPath := filepath.Join(werf.GetTmpDir(), "werf-ssh-agent", uuid.NewString())
+	var sockPath string
+	// darwin does not support path more than 108 characters
+	if runtime.GOOS == "darwin" {
+		sockPath = filepath.Join("/private/tmp/werf/Listeners", uuid.NewString())
+	} else {
+		sockPath = filepath.Join(werf.GetTmpDir(), "werf-ssh-agent", uuid.NewString())
```
Comment (alexey-igrychev) [inline pkg/ssh_agent/ssh_agent.go:206]: Could we solve it globally by changing the werf tmp dir?

**Review by alexey-igrychev — CHANGES_REQUESTED** [review]

File: docs/_data/werf_yaml.yml [hunk 5]
```diff
@@ -260,8 +260,11 @@ sections:
       - name: ssh
         value: "string"
         description:
-          en: SSH agent socket or keys to the build (similar to the docker build --ssh option)
-          ru: Сокет агента SSH или ключи для сборки (подобно docker build --ssh)
+          en: (deprecated) SSH agent socket. 
+          ru: (устарело) Сокет агента SSH. По умолчанию используется запущенный ssh агент или werf запустит временный агент.
```
Comment (alexey-igrychev) [inline docs/_data/werf_yaml.yml:264]:
```suggestion
          en: SSH agent socket or keys to the build (similar to the docker build --ssh option) (DEPRECATED)
          ru: Сокет агента SSH или ключи для сборки (подобно docker build --ssh) (DEPRECATED)
```

File: docs/pages_en/usage/build/images.md [hunk 6]
```diff
@@ -189,6 +189,21 @@ RUN --mount=type=secret,id=plainSecret \
     export WERF_BUILD_SECRET="$(cat /run/secrets/plainSecret)"
 ```
 
+### Using the SSH Agent
```
Comment (alexey-igrychev) [inline docs/pages_en/usage/build/images.md:192]:
```suggestion
### Using the SSH agent
```

File: pkg/container_backend/utils.go [hunk 7]
```diff
@@ -4,11 +4,18 @@ import (
...
 
 	"github.com/docker/docker/pkg/stringid"
 
 	"github.com/werf/logboek"
+	"github.com/werf/werf/v2/pkg/ssh_agent"
+)
+
+const (
+	SSHHostAuthSockPath      = "/run/host-services/ssh-auth.sock"
+	SSHContainerAuthSockPath = "/.werf/tmp/ssh-auth-sock"
```
[the remark this answers is not in the document]
Comment (iapershin) [reply pkg/container_backend/utils.go:18]: fixed. should be used here https://github.com/werf/werf/blob/08e44e57853379b97bfac866e9257fd6d2ee1257/pkg/container_backend/utils.go#L76C1-L77C59

**Pull request merged**